pyouroboros/dockerclient.py

```diff
@@ -187,127 +187,140 @@ def update(self):
+            else:
+                continue
+
+            # Get container list to restart after update complete
+            depends_on = container.labels.get('com.ouroboros.depends_on', False)
+            hard_depends_on = container.labels.get('com.ouroboros.hard_depends_on', False)
+            if depends_on:
+                depends_on_names.extend([name.strip() for name in depends_on.split(',')])
+            if hard_depends_on:
+                hard_depends_on_names.extend([name.strip() for name in hard_depends_on.split(',')])
+
+        hard_depends_on_containers = []
+        hard_depends_on_names = list(set(hard_depends_on_names))
+        for name in hard_depends_on_names:
+            try:
+                hard_depends_on_containers.append(self.client.containers.get(name))
+            except NotFound:
+                self.logger.error("Could not find dependant container %s on socket %s. Ignoring", name, self.socket)
+
+        depends_on_containers = []
+        depends_on_names = list(set(depends_on_names))
+        depends_on_names = [name for name in depends_on_names if name not in hard_depends_on_names]
+        for name in depends_on_names:
+            try:
+                depends_on_containers.append(self.client.containers.get(name))
+            except NotFound:
+                self.logger.error("Could not find dependant container %s on socket %s. Ignoring", name, self.socket)
+
+        return updateable, depends_on_containers, hard_depends_on_containers
+
+    def stop_container(self, container):
+        self.logger.debug('Stopping container: %s', container.name)
+        stop_signal = container.labels.get('com.ouroboros.stop_signal', False)
+        if stop_signal:
+            try:
+                container.kill(signal=stop_signal)
+            except APIError as e:
+                self.logger.error('Cannot kill container using signal %s. stopping normally. Error: %s',
+                                  stop_signal, e)
+                container.stop()
+        else:
+            container.stop()
+
+    def recreate(self, container, latest_image):
+        new_config = set_properties(old=container, new=latest_image)
+
+        self.stop_container(container)
+
+        self.logger.debug('Removing container: %s', container.name)
+        try:
+            container.remove()
+        except NotFound as e:
+            self.logger.error("Could not remove container. Error: %s", e)
+            return
+
+        created = self.client.api.create_container(**new_config)
+        new_container = self.client.containers.get(created.get("Id"))
+
+        # connect the new container to all networks of the old container
+        for network_name, network_config in container.attrs['NetworkSettings']['Networks'].items():
+            network = self.client.networks.get(network_config['NetworkID'])
+            try:
+                network.disconnect(new_container.id, force=True)
+            except APIError:
+                pass
```

Comment:
If the container can't be detached from the old network we can't attach it to the same network later for sure. We could already print an error here (like below `self.logger.error('Unable to attach updated container to network "%s". Error: %s', network, e)`) and `continue` the loop. It may be interesting to know for the user why the container can't be detached IMHO.

Reply:
the entire purpose of force=true is to ensure that _NO MATTER WHAT_ the container is detached. If it we didnt force it i would totally agree with you. But force is ensuring it does. If you can create an insance where it doesnt detach with force=true i would be open to it, but i dont believe its possible.

Reply:
That's true, got it. :slightly_smiling_face: